semgrep-core/src/parsing/other/yaml_to_generic.mli

```diff
@@ -1,3 +1,8 @@
+(* Note!!
+   `program` and `any` will behave slightly differently on metavariables.
```

Comment:
I understand this comment now, but in X months I probably don't. Some of the explanation that went into the PR comment should probably go here too.